src/backends/interflop-mca-mpfr/interflop_mca_mpfr.c

```diff
@@ -152,17 +157,23 @@ static int _mca_inexact(mpfr_ptr a, mpfr_rnd_t rnd_mode) {
   mpfr_clear(mpfr_zero);
 }
 
-static void _mca_seed(void) {
+static void _set_mca_seed(int choose_seed, int seed) {
   const int key_length = 3;
   uint64_t init_key[key_length];
-  struct timeval t1;
-  gettimeofday(&t1, NULL);
-
-  /* Hopefully the following seed is good enough for Montercarlo */
-  init_key[0] = t1.tv_sec;
-  init_key[1] = t1.tv_usec;
-  init_key[2] = getpid();
-
+  
+  if (choose_seed) {
+    init_key[0] = seed;
+    init_key[1] = seed;
+    init_key[2] = seed;    
```

Comment:
instead of doing this and then calling `tinymt64_init_by_array` you could directly call `void tinymt64_init(tinymt64_t * random, uint64_t seed);` which would be cleaner. The seed should be of type uint64_t.